### PR TITLE
test(sweeps): Add basic sweeps on launch regression test

### PR DESCRIPTION
**Pull request by TimH98**
Fixes
-----
- Fixes [WB-14563](https://wandb.atlassian.net/browse/WB-14563)

Description
-----------
Add a simple test running a sweep using a launch agent. An example resulting sweep from these tests can be seen [here](https://wandb.ai/launch-release-testing/release-testing/sweeps/l6wz0p9v)

Also refactor several utility functions to `utils.py`.

### <samp>🤖 Generated by Copilot at 812936d</samp>

This pull request enhances the test coverage and code quality of the launch agent's kubernetes integration. It adds new tests for running a sweep on a k8s cluster, and refactors some common utility functions in `utils.py` and `test_kubernetes.py`.

Testing
-------
Run `tox -e launch-release -- -k test_kubernetes_agent_on_local_process_sweep`

### <samp>🤖 Generated by Copilot at 812936d</samp>

> _To test the launch agent's k8s feature_
> _We added some tests and refactored_
> _We used `utils.py`_
> _To run sweeps and verify_
> _The output and job status were captured_

[WB-14563]: https://wandb.atlassian.net/browse/WB-14563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gtarpenning — COMMENTED** [review]
I am a huge fan of this, can't wait to add more tests :) Considering I contributed, I'm not sure I should also review.

File: tests/release_tests/test_launch/utils.py [hunk 1]
```diff
@@ -92,3 +110,61 @@ def update_dict(original, updated):
...
+    yaml.dump(
+        launch_agent_dict,
+        open("tests/release_tests/test_launch/launch-agent.yml", "w+"),
+    )
+
+
+def init_agent_in_launch_cluster(namespace: str):
+    """Build and deploy the agent in provided cluster namespace."""
+    create_config_files()
+    # Comment out if launch agent image is already built locally
```
Comment (KyleGoyette) [inline tests/release_tests/test_launch/utils.py:153]: Can we instead make an image argument for the release tests that run all the tests with a specific image? And only build if not provided?

Also the build step feels like it should be a step outside of each of the test steps and instead in the test setup (in pytests init fixture or in the tox steps)

**Review by KyleGoyette — COMMENTED** [review]
One issue with setup, but the new test is great!

File: tests/release_tests/test_launch/utils.py (hunk 1 above)
Comment (TimH98) [reply tests/release_tests/test_launch/utils.py]: Good idea, I moved the build step out to `pytest_configure` so it's only called once, and only if an image isn't provided.

**Review by KyleGoyette — APPROVED** [review]

File: tests/release_tests/test_launch/test_kubernetes.py [hunk 2]
```diff
@@ -56,49 +60,9 @@ def test_kubernetes_agent_on_local_process():
...
-    run_cmd(
-        "python tools/build_launch_agent.py --tag wandb-launch-agent:release-testing"
-    )
-    run_cmd("kubectl apply -f tests/release_tests/test_launch/launch-config.yml")
-    run_cmd("kubectl apply -f tests/release_tests/test_launch/launch-agent.yml")
-
-    setup_cleanup_on_exit(NAMESPACE)
-
+def test_kubernetes_agent_in_cluster(api_key: str, agent_image: Optional[str]):
+    init_agent_in_launch_cluster(NAMESPACE, api_key, agent_image)
```
[the remark this answers is not in the document]
Comment (TimH98) [reply tests/release_tests/test_launch/test_kubernetes.py:65]: Considering it's all self-contained, does the namespace matter? I don't see the advantage of making it an arg.

File: tests/release_tests/test_launch/launch-config-patch.yaml [hunk 3]
```diff
@@ -40,4 +40,9 @@ data:
...
     environment:
       type: aws
-      region: us-east-2
\ No newline at end of file
+      region: us-east-2
+---
+metadata:
+  namespace: wandb-release-testing
+stringData:
+  password: "" # Placeholder
```
[the remark this answers is not in the document]
Comment (TimH98) [reply tests/release_tests/test_launch/launch-config-patch.yaml:48]: We fill in the password with the wandb api key [here](https://github.com/wandb/wandb/blob/ed08ef3dfeddef5725c7415f11267ec96a52dad2/tests/release_tests/test_launch/utils.py#L130). It's not meant to be changed directly by the user, I'll make that clearer.

File: tests/release_tests/test_launch/test_kubernetes.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TimH98) [reply tests/release_tests/test_launch/test_kubernetes.py:65]: Either way, I made https://wandb.atlassian.net/browse/WB-15487 but IMO it's very low priority

**Pull request merged**